File: editor/en/[1544]Make The String Great.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def makeGood(self, s: str) -> str:
        stack = []

        for char in s:
            if stack and stack[-1].lower() == char.lower() and stack[-1] != char:
                stack.pop()
            else:
                stack.append(char)

        return ''.join(stack)

# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1544]Make The String Great.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function (s) {
    const stack = []

    for (let char of s) {
        const topChar = stack[stack.length - 1];

        if (stack.length && char.toLowerCase() === topChar.toLowerCase() && char !== topChar) {
            stack.pop();
        } else {
            stack.push(char);
        }
    }

    return stack.join('');
};
//leetcode submit region end(Prohibit modification and deletion)
